documents/src/model.ts

```diff
@@ -0,0 +1,366 @@
+   * or documents that use kernels, and they store the name and the language of the kernel.
+   */
+  readonly defaultKernelName: string;
+  readonly defaultKernelLanguage: string;
+
+  /**
+   * modelBD is the datastore for the content of the document.
+   * modelDB is not a shared datastore so we don't use it on this example since
+   * this example is a shared document.
+   */
+  readonly modelDB: IModelDB;
+
+  /**
+   * New datastore introduced in JupyterLab v3.1 to store shared data and make notebooks
+   * collaborative
+   */
+  readonly sharedModel: ExampleDoc = ExampleDoc.create();
+
+  /**
+   * Dispose of the resources held by the model.
+   */
+  dispose(): void {
+    if (this._isDisposed) {
+      return;
+    }
+    this._isDisposed = true;
+    Signal.clearData(this);
+  }
+
+  /**
+   * Should return the data that you need to store in disk as a string.
+   * The context will call this method to get the file's content and save it
+   * to disk
+   */
+  toString(): string {
+    const pos = this.sharedModel.getContent('position');
+    const obj = {
+      x: pos?.x || 10,
+      y: pos?.y || 10,
+      content: this.sharedModel.getContent('content') || '',
+    };
+    return JSON.stringify(obj, null, 2);
+  }
+
+  /**
+   * The context will call this method when loading data from disk.
+   * This method should implement the logic to parse the data and store it
+   * on the datastore.
+   *
+   * @param data
+   */
+  fromString(data: string): void {
+    const obj = JSON.parse(data);
+    this.sharedModel.transact(() => {
+      this.sharedModel.setContent('position', { x: obj.x, y: obj.y });
+      this.sharedModel.setContent('content', obj.content);
+    });
+  }
+
+  /**
+   * Should return the data that you need to store in disk as a JSON object.
+   * The context will call this method to get the file's content and save it
+   * to disk.
+   *
+   * NOTE: This method is only used by the context of the notebook, every other
+   * document will load/save the data through toString/fromString.
+   */
+  toJSON(): PartialJSONObject {
+    const pos = this.sharedModel.getContent('position');
+    const obj = {
+      x: pos?.x || 10,
+      y: pos?.y || 10,
+      content: this.sharedModel.getContent('content') || '',
+    };
+    return obj;
+  }
+
+  /**
+   * The context will call this method when loading data from disk.
+   * This method should implement the logic to parse the data and store it
+   * on the datastore.
+   *
+   * NOTE: This method is only used by the context of the notebook, every other
+   * document will load/save the data through toString/fromString.
+   *
+   * @param data
+   */
+  fromJSON(data: PartialJSONObject): void {
+    this.sharedModel.transact(() => {
+      this.sharedModel.setContent('position', { x: data.x, y: data.y });
+      this.sharedModel.setContent('content', data.content);
+    });
+  }
+
+  /**
+   *
+   */
+  initialize(): void {
+    // not implemented
```

Comment:
May be that is a more appropriate wording - _not implemented_ sounds like you should have something here but you forget to do it. This definitely should not append in examples.

```suggestion
    // nothing to do
```